Steps4Impact/DashboardV2/DashboardViewController.swift

```diff
@@ -62,10 +64,40 @@ class DashboardViewController: TableViewController {
     }
   }
 
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+    askForNotificationPermission()
+  }
+
   deinit {
     NotificationCenter.default.removeObserver(self)
   }
 
+  private func askForNotificationPermission() {
+    UNUserNotificationCenter.current().getNotificationSettings { (settings) in
+      DispatchQueue.main.async {
+        if settings.authorizationStatus == .authorized {
+          AppController.shared.registerForRemoteNotifications()
+        }
+        guard settings.authorizationStatus == .notDetermined else {
+          return
+        }
+        let controller = UIAlertController(
+          title: Strings.NotificationsPermission.title,
+          message: Strings.NotificationsPermission.message, preferredStyle: .alert)
+        controller.addAction(UIAlertAction(
+          title: Strings.NotificationsPermission.proceed, style: .default,
+          handler: { (_) in
```

Comment:
The tuple in the closure is unnecessary, just blackhole the argument.  Extracting the action with the trailing closure is nicer.